homeassistant/components/wemo/wemo_device.py

```diff
@@ -0,0 +1,89 @@
+"""Home Assistant wrapper for a pyWeMo device."""
+import logging
+
+from pywemo import PyWeMoException, WeMoDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_DEVICE_ID,
+    CONF_NAME,
+    CONF_PARAMS,
+    CONF_TYPE,
+    CONF_UNIQUE_ID,
+)
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN, WEMO_SUBSCRIPTION_EVENT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DeviceWrapper:
+    """Home Assistant wrapper for a pyWeMo device."""
+
+    def __init__(self, hass: HomeAssistant, wemo: WeMoDevice, device_id: str) -> None:
+        """Initialize DeviceWrapper."""
+        self.hass = hass
+        self.wemo = wemo
+        self.device_id = device_id
+        self.device_info = _device_info(wemo)
+        self.supports_long_press = wemo.supports_long_press()
+
+    def _subscription_callback(
+        self, _device: WeMoDevice, event_type: str, params: str
+    ) -> None:
+        self.hass.bus.fire(
+            WEMO_SUBSCRIPTION_EVENT,
+            {
+                CONF_DEVICE_ID: self.device_id,
+                CONF_NAME: self.wemo.name,
+                CONF_TYPE: event_type,
```

Comment:
What events can this fire? Normally we don't want to fire events for changes that are already handled by an entity.

Reply:
Changed to only fire events not handled by the entity.